alvr/client_openxr/src/interaction.rs

```diff
@@ -43,223 +65,222 @@ pub struct BodySources {
+        ));
 
+        bindings.push(binding(&left_aim_action, "/user/hand/left/input/aim/pose"));
         bindings.push(binding(
-            &left_detached_controller_pose_action,
-            "/user/detached_controller_meta/left/input/grip/pose",
+            &right_aim_action,
+            "/user/hand/right/input/aim/pose",
         ));
+
         bindings.push(binding(
-            &right_detached_controller_pose_action,
-            "/user/detached_controller_meta/right/input/grip/pose",
+            &left_vibration_action,
+            "/user/hand/left/output/haptic",
+        ));
+        bindings.push(binding(
+            &right_vibration_action,
+            "/user/hand/right/output/haptic",
         ));
-    }
 
-    // Apply bindings:
-    xr_ctx
-        .instance
-        .suggest_interaction_profile_bindings(
-            xr_ctx
-                .instance
-                .string_to_path(controllers_profile_path)
-                .unwrap(),
-            &bindings,
-        )
-        .unwrap();
+        let left_detached_controller_pose_action;
+        let right_detached_controller_pose_action;
+        if supports_multimodal {
+            // Note: when multimodal input is enabled, both controllers and hands will always be active.
+            // To be able to detect when controllers are actually held, we have to register detached
+            // controllers pose; the controller pose will be diverted to the detached controllers when
+            // they are not held. Currently the detached controllers pose is ignored
+            left_detached_controller_pose_action = action_set
+                .create_action::<xr::Posef>(
+                    "left_detached_controller_pose",
+                    "Left detached controller pose",
+                    &[],
+                )
+                .unwrap();
+            right_detached_controller_pose_action = action_set
+                .create_action::<xr::Posef>(
+                    "right_detached_controller_pose",
+                    "Right detached controller pose",
+                    &[],
+                )
+                .unwrap();
+
+            bindings.push(binding(
+                &left_detached_controller_pose_action,
+                "/user/detached_controller_meta/left/input/grip/pose",
+            ));
+            bindings.push(binding(
+                &right_detached_controller_pose_action,
+                "/user/detached_controller_meta/right/input/grip/pose",
+            ));
+        }
+
+        // Apply bindings:
+        xr_instance
+            .suggest_interaction_profile_bindings(
+                xr_instance
+                    .string_to_path(controllers_profile_path)
+                    .unwrap(),
+                &bindings,
+            )
+            .unwrap();
+
+        let combined_eyes_source = if xr_instance.exts().ext_eye_gaze_interaction.is_some()
+            && !platform.is_quest()
+            && !platform.is_vive()
+        {
```

Comment:
is the other extension preferred for eye tracking on the quest pro?

Reply:
Well not officially, but the fb extension provides two eyes instead of one gaze which is more info for the downstream applications. Well, actually the quest pro returns fixed vergence (at around the distance of people would talk to face to face). So it's not true we get more info, but hopefully they can improve on this without changing the api. But the main reason is eye gaze interaction can be enabled only at session startup so we don't get a second chance later when we connect to the server

Reply:
So no drawback occurs when using this extension instead. That's good enough.